diff --git a/src/mac/carbon/textctrl.cpp b/src/mac/carbon/textctrl.cpp
index d116f73..1fd6446 100644
--- a/src/mac/carbon/textctrl.cpp
+++ b/src/mac/carbon/textctrl.cpp
@@ -18,7 +18,7 @@
 #if wxUSE_TEXTCTRL
 
 
-#ifdef __DARWIN__
+#if defined(__DARWIN__) || defined(Macintosh)
   #include <sys/types.h>
   #include <sys/stat.h>
 #else
@@ -63,7 +63,7 @@
 #define wxMAC_AWAYS_USE_MLTE 1
 #endif
 
-#ifndef __WXMAC_OSX__
+#if !(defined(__WXMAC_OSX__) || defined(Macintosh))
 enum
 {
     kTXNVisibilityTag             = 'visb' /*set the visibility state of the object  */
diff --git a/src/common/file.cpp b/src/common/file.cpp
index ae71d0e..857c829 100644
--- a/src/common/file.cpp
+++ b/src/common/file.cpp
@@ -57,7 +57,7 @@
     #include  "wx/msw/missing.h"
 #elif (defined(__OS2__))
     #include <io.h>
-#elif (defined(__UNIX__) || defined(__GNUWIN32__))
+#elif (defined(__UNIX__) || defined(__GNUWIN32__) || defined(Macintosh))
     #include  <unistd.h>
     #include  <time.h>
     #include  <sys/stat.h>
@@ -105,7 +105,7 @@
 
 // there is no distinction between text and binary files under Unix, so define
 // O_BINARY as 0 if the system headers don't do it already
-#if defined(__UNIX__) && !defined(O_BINARY)
+#if (defined(__UNIX__) || defined(Macintosh)) && !defined(O_BINARY)
     #define   O_BINARY    (0)
 #endif  //__UNIX__
 
@@ -438,7 +438,7 @@ bool wxFile::Eof() const
 
     wxFileOffset iRc;
 
-#if defined(__DOS__) || defined(__UNIX__) || defined(__GNUWIN32__) || defined( __MWERKS__ ) || defined(__SALFORDC__)
+#if defined(__DOS__) || defined(__UNIX__) || defined(__GNUWIN32__) || defined( __MWERKS__ ) || defined(__SALFORDC__) || defined(Macintosh)
     // @@ this doesn't work, of course, on unseekable file descriptors
     wxFileOffset ofsCur = Tell(),
     ofsMax = Length();
diff --git a/src/common/filename.cpp b/src/common/filename.cpp
index a0523bb..1646f5f 100644
--- a/src/common/filename.cpp
+++ b/src/common/filename.cpp
@@ -101,7 +101,7 @@
 #endif
 
 // utime() is POSIX so should normally be available on all Unices
-#ifdef __UNIX_LIKE__
+#if defined(__UNIX_LIKE__) || defined(Macintosh)
 #include <sys/types.h>
 #include <utime.h>
 #include <sys/stat.h>
diff --git a/src/mac/carbon/dirmac.cpp b/src/mac/carbon/dirmac.cpp
index bec8035..c3680d0 100644
--- a/src/mac/carbon/dirmac.cpp
+++ b/src/mac/carbon/dirmac.cpp
@@ -36,7 +36,7 @@
 #include "wx/dir.h"
 #include "wx/filefn.h"          // for wxDirExists()
 
-#ifndef __DARWIN__
+#if !(defined(__DARWIN__) || defined(Macintosh))
   #include <windows.h>
 #endif
 
diff --git a/include/wx/mac/carbon/checklst.h b/include/wx/mac/carbon/checklst.h
index df3f235..085e7a3 100644
--- a/include/wx/mac/carbon/checklst.h
+++ b/include/wx/mac/carbon/checklst.h
@@ -17,7 +17,7 @@
 #pragma interface "checklst.h"
 #endif
 
-#if !defined(__MWERKS__) && !defined(__UNIX__)
+#if !defined(__MWERKS__) && !defined(__UNIX__) && !defined(Macintosh)
 typedef   unsigned int  size_t;
 #endif
 
diff --git a/src/mac/carbon/app.cpp b/src/mac/carbon/app.cpp
index 703f001..d679366 100644
--- a/src/mac/carbon/app.cpp
+++ b/src/mac/carbon/app.cpp
@@ -43,6 +43,10 @@
 
 // mac
 
+#ifdef Macintosh
+  #define __option(x) 0
+#endif
+
 #ifndef __DARWIN__
   #if __option(profile)
     #include <profiler.h>
