diff --git a/src/mac/carbon/morefile/MoreFilesExtras.c b/src/mac/carbon/morefile/MoreFilesExtras.c
index f9bda4d..9788e74 100644
--- a/src/mac/carbon/morefile/MoreFilesExtras.c
+++ b/src/mac/carbon/morefile/MoreFilesExtras.c
@@ -494,7 +494,7 @@ pascal	void	TruncPString(StringPtr destination,
 
 /*****************************************************************************/
 
-static pascal	Ptr	GetTempBuffer(long buffReqSize,
+pascal	Ptr	GetTempBuffer(long buffReqSize,
 						  long *buffActSize)
 {
 	enum
