diff --git a/include/wx/socket.h b/include/wx/socket.h
index 15a718b..2c57d72 100644
--- a/include/wx/socket.h
+++ b/include/wx/socket.h
@@ -128,12 +128,12 @@ public:
 
   // base IO
   virtual bool  Close();
-  wxSocketBase& Discard();
+  virtual wxSocketBase& Discard();
   wxSocketBase& Peek(void* buffer, wxUint32 nbytes);
-  wxSocketBase& Read(void* buffer, wxUint32 nbytes);
+  virtual wxSocketBase& Read(void* buffer, wxUint32 nbytes);
   wxSocketBase& ReadMsg(void *buffer, wxUint32 nbytes);
   wxSocketBase& Unread(const void *buffer, wxUint32 nbytes);
-  wxSocketBase& Write(const void *buffer, wxUint32 nbytes);
+  virtual wxSocketBase& Write(const void *buffer, wxUint32 nbytes);
   wxSocketBase& WriteMsg(const void *buffer, wxUint32 nbytes);
 
   void InterruptWait() { m_interrupt = true; };
