diff --git a/wxc/include/wrapper.h b/wxc/include/wrapper.h
index 8470093..2065ad3 100644
--- a/wxc/include/wrapper.h
+++ b/wxc/include/wrapper.h
@@ -128,7 +128,7 @@ class ELJDataObject: public wxObject
                 ELJDataObject(void* _data) : wxObject() {data = _data;};
                 void* data;
 };
-
+#if wxUSE_DRAG_AND_DROP
 class ELJDropTarget : public wxDropTarget
 {
         private:
@@ -249,7 +249,7 @@ class ELJFileDropTarget : public wxFileDropTarget
                 void SetOnDragOver (DragThreeFunc _func) {on_drag_func = _func;};
                 void SetOnLeave    (DragZeroFunc _func)  {on_leave_func = _func;};
 };
-
+#endif //wxUSE_DRAG_AND_DROP
 class ELJTextValidator : public wxTextValidator
 {
         public:
@@ -271,7 +271,7 @@ class ELJTextValidator : public wxTextValidator
                 void*        obj;
                 ValidateFunc fnc;
 };
-
+#if wxUSE_SOCKETS
 class ELJConnection : public wxTCPConnection
 {
         private:
@@ -385,7 +385,7 @@ class ELJClient : public wxTCPClient
                         return result;
                 };
 };
-
+#endif // wxUSE_SOCKETS
 class ELJPrintout : public wxPrintout
 {
         private:
