--- a/OpenTransport.h        1970-01-01 01:00:01.000000000 +0100
+++ b/OpenTransport.h        2025-06-11 23:04:07.124320089 +0200
@@ -17,9 +17,46 @@
 #ifndef __OPENTRANSPORT__
 #define __OPENTRANSPORT__
 
-
-
-
+#undef SIGHUP
+#undef SIGPOLL
+#undef SIGURG
+#if OTUNIXERRORS
+#undef EWOULDBLOCK
+#undef EALREADY
+#undef ENOTSOCK
+#undef EDESTADDRREQ
+#undef EMSGSIZE
+#undef EPROTOTYPE
+#undef ENOPROTOOPT
+#undef EPROTONOSUPPORT
+#undef EOPNOTSUPP
+#undef EADDRINUSE
+#undef EADDRNOTAVAIL
+#undef ENETDOWN
+#undef ENETUNREACH
+#undef ENETRESET
+#undef ECONNABORTED
+#undef ECONNRESET
+#undef ENOBUFS
+#undef EISCONN
+#undef ENOTCONN
+#undef ESHUTDOWN
+#undef ETOOMANYREFS
+#undef ETIMEDOUT
+#undef ECONNREFUSED
+#undef EHOSTDOWN
+#undef EHOSTUNREACH
+#undef EPROTO
+#undef ETIME
+#undef ENOSR
+#undef EBADMSG
+#undef ECANCEL
+#undef ENOSTR
+#undef ENODATA
+#undef EINPROGRESS
+#undef ESRCH
+#undef ENOMSG
+#endif
 /*
    The following table shows how to map from the old (pre-Universal
    Interfaces) header file name to the equivalent Universal Interfaces
